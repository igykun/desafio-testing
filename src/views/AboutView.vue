<template>
    <div>
        
    </div>
</template>
<script>
export default {
    name: 'AboutView',
    data(){
        return{
            
        }
    }
}
</script>
<style>
    
</style>