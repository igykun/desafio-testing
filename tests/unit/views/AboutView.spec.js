import { mount } from "@vue/test-utils";
import AboutView from "@/views/AboutView.vue";
import { createRouter, createWebHistory } from "vue-router";

describe('AboutView', () => { 
    test('Probando la existencia del componente o vista AboutView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/about',
                name: 'AboutViewVue',
                component: AboutView
            }],
        })
        router.push('/about')
        await router.isReady()

        const wrapper = mount(AboutView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(AboutView).exists()).toBe(true)
    })

    test('Genera un snapshot de la vista AboutView', async ()=>{
        const router = createRouter({
            history:createWebHistory(),
            routes:[{
                path: '/about',
                name: 'about',
                component: AboutView
            }],
        })
        router.push('/about')
        await router.isReady()
        const wrapper=mount(AboutView,{
            global:{
                plugins:[router]
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })
})