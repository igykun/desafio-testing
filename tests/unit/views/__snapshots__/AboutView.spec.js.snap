// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView Genera un snapshot de la vista AboutView 1`] = `<div></div>`;
